src/newgrf_gui.cpp

```diff
@@ -1045,13 +1049,15 @@ struct NewGRFWindow : public Window, NewGRFScanCallback {
 				this->avail_pos = -1;
 				this->avail_sel = nullptr;
 				this->avails.ForceRebuild();
+				this->modified = true;
 				this->InvalidateData(GOID_NEWGRF_LIST_EDITED);
```

Comment:
As far as I can see, you add the `modified = true` line to all `GOID_NEWGRF_LIST_EDITED`, so why not change it directly in `OnInvalidateData`?